dask/dataframe/groupby.py

```diff
@@ -671,7 +671,10 @@ def _groupby_apply_funcs(df, *index, **kwargs):
         else:
             result[result_column] = r
 
-    return pd.DataFrame(result)
+    if is_dataframe_like(df):
+        return type(df)(result)
+    else:
+        return type(df.head(0).to_frame())(result)
```

Comment:
What type is this catching? Could you add a comment explaining what may show up here?

Reply:
Trying to convert Series-like objects into DataFrames.  I've added a comment in the latest commit.

Reply:
Thanks Matt.